File: app/src/main/java/il/ronmad/speedruntimer/Point.java
package il.ronmad.speedruntimer;

import io.realm.RealmObject;

public class Point extends RealmObject {

    public int x;
    public int y;

    void set(int x, int y) {
        getRealm().executeTransaction(realm -> {
            this.x = x;
            this.y = y;
        });
    }
}

File: app/src/main/kotlin/il/ronmad/speedruntimer/Point.kt
package il.ronmad.speedruntimer

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

open class Point : RealmObject(), HasPrimaryId {

    @PrimaryKey
    override var id: Long = 0L

    var x: Int = 0
    var y: Int = 0

    operator fun component1() = this.x

    operator fun component2() = this.y
}
